test/Microsoft.AspNet.TestHost.Tests/ClientHandlerTests.cs

```diff
@@ -269,5 +275,61 @@ public Task ProcessRequestAsync(Context context)
                 return _application(context.HttpContext);
             }
         }
+
+
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Hangs randomly (issue #507)")]
+        public async Task ClientHandlerCreateContextWithDefaultRequestParameters()
+        {
+            var logger = new VerifierLogger();
+            var builder = new WebApplicationBuilder()
+                            .ConfigureServices(services =>
+                            {
+                                services.AddSingleton<ILogger<WebApplication>>(logger);
+                            })
+                            .Configure(app =>
+                            {
+                                app.Run(context =>
+                                {
+                                    return Task.FromResult(0);
+                                });
+                            });
+            var server = new TestServer(builder);
+
+            var result = await server.CreateClient().GetStringAsync("/");
+            Assert.Contains("Request starting HTTP/1.1 GET http://", logger.Message);
```

Comment:
We don't even need the assert at all. We can regard the test as successful as long as no exception was thrown during context creation. I added the assert as a sanity check that at least the logger is able to extract some reasonable values. 

Asserting non-empty is not very useful since `HostingLogScope` and `HostingRequestFinished` messages will almost always guarantee that it's non-empty. The test already looks strange as it is, having no assert or simply asserting it's not empty seems to be even more confusing.

Though maybe I should add some comments to at least indicate what the test is actually doing...